### PR TITLE
Fix: Update style for member and token popovers

### PR DESCRIPTION
**Pull request by mmioana**
## Description

Add domain color to ContributorReputation and ContributorRoles fragments
Update styling for user popover
Update styling for token popover

## Testing

* Step 1. Go to colony members
* Step 2. Check the user card styling for each type of member (Top/Dedicated/Verified/Unverified etc) - 
  * scrollable Teams section with a maximum of 4 teams displayed 
Screenshot 2024-07-04 at 09 11 51

  * team pills in upper section have a tooltip if text is truncated
Screenshot 2024-07-04 at 09 11 42

  * card padding for mobile and scrollbar
* Step 3. Go to colony finance
* Step 4. Check the token card styling - different width for tablet/desktop

Resolves https://github.com/JoinColony/colonyCDapp/issues/2496

### CODE REVIEW

**Comment by mmioana** [comment]
@iamsamgibbs @melyndav I have opened another PR for https://github.com/JoinColony/colonyCDapp/issues/2496 containing all needed styling changes requested in PR #2586 as I rebased the wrong branch

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @mmioana - continuing comments from (so I have a record) #2586 

1. I'm getting some issues with the different contributor card types: 

image

image

image

image

However, this issue seems to be local to the 'Contributor' tab and not members since the other tab is showing fine. 

image

image


2. Also it seems the about section has been set with a fixed height on desktop only but hasn't been carried over to mobile. Please remove this since max height since we have a max char limit set in the user account area. This long text is only for the dev env. 

image


3. Mobile is missing padding at the bottom of the cards, clipping into the team section:

image

File: src/components/v5/common/Pills/PillsBase.tsx [hunk 1]
```diff
@@ -14,33 +14,39 @@ const PillsBase: FC<PropsWithChildren<PillsProps>> = ({
...
+      className={clsx(
+        'inline-flex shrink-0 items-center rounded-3xl px-2.5 py-1.5 text-center',
+        className,
+        {
+          capitalize: isCapitalized,
+          'h-[1.625rem] text-3': pillSize === 'medium',
+          'h-[1.5rem] text-4': pillSize === 'small',
+        },
+      )}
+      {...rest}
```
Comment (Nortsova) [inline src/components/v5/common/Pills/PillsBase.tsx:31]: Hi there, thanks for your PR! Just a quick note: I noticed something about props spreading and types that we could improve. There are some props in the type that aren't being extracted from props, which means they might end up as attributes in the <span> element by accident.

Here are the props that could be spread into <span>:

```
export interface PillsProps {
  mode?: ExtensionStatusBadgeMode | UserStatusMode;
  iconSize?: IconSize;
  isIconVisible?: boolean;
}
```

**Comment by mmioana** [comment]
@melyndav @Nortsova I did the requested changes, if you want please to have another check 🚀 💮

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi again @mmioana - looks like we're getting closer with this, however, some further changes are needed.

It seems the content container is not matching Figma in regard to width and as a result is truncating the team name. These boxes were designed to handle the max char on team names without truncation needing to happen.

image

Even with longer rep values, the team should not truncate.

image
 
Be sure to check all card variants to ensure they don't truncate as it's happening not just on the top contributor.

image

It could be caused by the scroll, however the internal contain container width does not match Figma's px by 2. 

image

**Review by melyndav — CHANGES_REQUESTED** [review]
Thanks for the continued updates @mmioana.

1. I'm still getting truncation on team names, as per my previous comment.

image

It seems this team container is smaller than the content container when in Figma it's the same.

Currently:
image

Figma:
image



2. For the token pop over, can you add the missing horizontal rule that's in Figma.

image


image

3. Also missing is the native token text for the native token in the Colony.

image

**Comment by mmioana** [comment]
@melyndav amazing eyes, I have definitely missed the token horizontal rule and text! 👀 
Hope now everything is in place, if you can please re-check 💮

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @mmioana 

1. Scroll bars have been introduced to modals in this PR -

image

On production and master (screenshot below) these are not showing.

image


2. I'm still seeing the teams being truncated when a second permission is applied. Since the container width remains the same with as per my previous comment. 

image

3. The native token pop over styling is still missing from my previous comment, including the HR and the copy.

image

If you can kindly double-check everything before requesting the next review and also doing a quick sanity check of the app, that would be helpful with reducing the time of the PR review process.

**Comment by iamsamgibbs** [comment]
Just a note to say I am experiencing this issue on master (and on the multi-sig branches) so if you come across it, it hasn't been introduced by a change here. (I am opening a new issue for it)

Screenshot 2024-07-09 at 12 22 15

**Comment by rdig** [comment]
What's the state here? Can this be reviewed or are we waiting to fix it on `master` then bring over the changes ?

Or are we just fixing everything, including stuff not relevant to the current issue in here ?

**Comment by mmioana** [comment]
@rdig I put it in draft state only to re-check if all needed changes for this issue are in place.
For the bug Sam discovered, there is another issue https://github.com/JoinColony/colonyCDapp/issues/2676

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @mmioana, I'm unable to test this any more. No matter how many times I restart with a fresh env, I just can't get to a test Colony. I have no problems on other branches or master etc. Any thoughts on what this could be? 

Export-1720789406248

**Comment by rdig** [comment]
@melyndav this is definitely a issue on your end, as I just tested this branch and everything is working fine.

What's the error you're getting in the console when the page crashes ?

Also try the following:
- remove the branch locally (`git branch -D <branch-name>`)
- pull the fresh branch again (`git fetch origin <branch-name> && git checkout <branch-name>`)
- remove node modules (`rm -R node_modules`)
- reinstall modules (`npm ci`)
- re-run the dev environment (`npm run dev`)

I'll submit my review in a bit, where you can see me actually being able to run this branch without issues

**Review by rdig — APPROVED** [review]
Tested all the cases I could think of, and all that were mentioned in the original issue, and as far as I can tell everything is as expected.

Screenshot from 2024-07-12 17-13-50
Screenshot from 2024-07-12 17-15-58
Screenshot from 2024-07-12 17-16-04
Screenshot from 2024-07-12 17-16-19
Screenshot from 2024-07-12 17-16-23
Screenshot from 2024-07-12 17-16-51
Screenshot from 2024-07-12 17-17-02
Screenshot from 2024-07-12 17-17-49
Screenshot from 2024-07-12 17-17-52
Screenshot from 2024-07-12 17-18-04
Screenshot from 2024-07-12 17-18-08

As for the environment not working, here's a short video of me running this locally

[Screencast from 2024-07-12 17-22-11.webm](https://github.com/user-attachments/assets/f09df2cb-617b-4a84-852f-01ae42505f08)

**Review by jakubcolony — APPROVED** [review]
I was able to run this branch without issues.

Nice work on this @mmioana, all the cases I tested look good:
image

image

image

image

image

image

image

image

----

The only problem I spotted is the reputation being shown in the wrong units in the tooltip. However, this is also present in `master`, so I opened #2706 to track.
image

**Review by iamsamgibbs — APPROVED** [review]
Amazing work on this @mmioana ! Fantastic job covering all these cases. All previously raised issues have been resolved. I check the user popovers in as many places as I could find, with different variations of permissions and teams. Checked the token popover too.

Screenshot 2024-07-16 at 10 14 45

Screenshot 2024-07-16 at 10 14 59

Screenshot 2024-07-16 at 10 15 26

Screenshot 2024-07-16 at 10 20 47

Nice job implementing this "click to open tooltip".

https://github.com/user-attachments/assets/dfa89b91-ccc4-4c33-a487-275f26392e3c

Screenshot 2024-07-16 at 10 26 36

Screenshot 2024-07-16 at 10 27 39

Screenshot 2024-07-16 at 10 28 02

Screenshot 2024-07-16 at 10 28 27

Screenshot 2024-07-16 at 10 32 27

I think I've seen this raised on another PR, but the 'verified' icon here should actually be the 'lock' icon. (@melyndav correct me if I'm wrong if it needs to be changed here)

Screenshot 2024-07-16 at 10 31 12

Screenshot 2024-07-16 at 10 42 14

**Comment by iamsamgibbs** [comment]
Ah yep, the locked icon is being tracked in #2667 so no need to fix here :)

**Comment by rdig** [comment]
@mmioana can you please rebase this ?

**Comment by mmioana** [comment]
@rdig on my way 🚀

**Comment by rdig** [comment]
@melyndav do want to review this still?

Otherwise I'll merge it since it's been open for quite a while

**Review by melyndav — APPROVED** [review]
Hey @mmioana! Was able to review this one again and everything now looks aligned with Figma. Nice work on a long PR! Appreciate your effort 🌼

**Pull request merged**